src/entities/offerings.ts

```diff
@@ -198,16 +270,47 @@ const toPrice = (priceData: { amount: number; currency: string }): Price => {
   };
 };
 
+const toPurchaseOptionPrice = (
+  optionPrice: PurchaseOptionPriceResponse,
+): PurchaseOptionPrice => {
+  return {
+    periodDuration: optionPrice.period_duration,
+    cycleCount: optionPrice.cycle_count,
+    price: optionPrice.price ? toPrice(optionPrice.price) : null,
+  } as PurchaseOptionPrice;
+};
+
+const toSubscriptionPurchaseOption = (
+  option: SubscriptionPurchaseOptionResponse,
+): SubscriptionPurchaseOption => {
+  return {
+    id: option.id,
+    basePrice: toPurchaseOptionPrice(option.base_price),
+    trial: option.trial ? toPurchaseOptionPrice(option.trial) : null,
+  } as SubscriptionPurchaseOption;
+};
+
 const toProduct = (
   productDetailsData: ProductResponse,
   presentedOfferingIdentifier: string,
 ): Product => {
+  const options: { [optionId: string]: SubscriptionPurchaseOption } = {};
+
+  Object.entries(productDetailsData.subscription_purchase_options).forEach(
+    ([key, value]) => {
+      options[key] = toSubscriptionPurchaseOption(value);
+    },
+  );
+
   return {
     identifier: productDetailsData.identifier,
     displayName: productDetailsData.title,
     currentPrice: toPrice(productDetailsData.current_price),
     normalPeriodDuration: productDetailsData.normal_period_duration,
```

Comment:
I think for the `currentPrice` and `normalPeriodDuration` maybe we should default to the value of the defaultSubscriptionOption and update the docs on these 2? Seems like the backend returned values should be deprecated (kept only for backwards compatibility). 

This might not work for in app products though, once we add them. So I'm thinking for this version to filter out responses where there is no `subscription_purchase_options` field in the backend response (and add a log). That way, once we add support for one time products, we don't break the parsing on these older SDKs. 

Separately, it might be good to provide an easy accessor for the trial info, even if it's only sintactic sugar. Otherwise it's more difficult to check whether there is a trial.